### ISSUE TITLE
Enrich and Clean the model library

### ISSUE THREAD

**Issue by ptaillandier** [gama-platform/gama#1345]
The model library is an importance tool for people to understand how to use GAMA features (and do some demos). However, it could be greatly improved:
 - Be sure that all models are written with GAMA 1.7 "style"
 - Choose a new name for each project that allows to simply understand on what the models inside are about.
 - Complete the header and **comments** all the models ! In particular for the ones in the features folder
 - In the features folder, remove the models that are too complex or decompose them into several simple models
 - All the features of GAMA should have one or several feature models explaining how to use it.

**Comment by AlexisDrogoul** [comment]
Double assignment to @ptaillandier  :)

**Comment by AlexisDrogoul** [comment]
Hi,

Feel free to remove outdated models (some of them here since the beginning of GAMA…). We might also want to include new models (I'm thinking about the nice "Urban Growth" model in its two flavors, raster-based and vector-based).

Alexis

**Comment by AlexisDrogoul** [comment]
I'd like to link this issue to the (now closed) enhancement request in #670. Basically, "features" models can now be declared in their own plugin, which will maybe help in having more examples for the additions made by plugins. Feel free to comment, also, on this issue.

**Comment by mazarsju** [comment]
Before getting into this task, we have to think about a common structure for all that (thinking also about the future website).

**Application**

For me, in the application view, it would be fine to add a folder (under each project folder) named "description" which contains an html page created automatically (by a future script in the documentation plugin). One html page for each model. This page has to contain : a title, a short description, the author name, and a screenshot images (one for each experiment). 
Those information can be found in comments, at the very beginning of each gaml file, according to the following organization:
```
/**
* Name: Here_The_Name
* Author: Author_Name
* Description: Description
* Tag: Tags_For_Website
*/
```
The screenshot images can be found in a "screenshot" folder, under each project folder (this folder exists in the github version, but not in the release version, because it is not very useful and quite heavy for the user).

**Website**

In the website, we can choose to sort the models by categories (ie : sort by feature, or sort by application domains), determined with the tag we put in it. For each model, the user can either display the description of the model, or the gaml code itself.

---------------------

To summarize the things to do : 
- Write a header with name, author, description, and tags for each models
- Create a folder "screenshot" for each projects (to see : maybe this task can be automatized ? Taking a screenshot for each experiment for the step number n ?)
- The documentation generation has to be extended in order to generate html pages
- A script have to be implemented to load the website database with the model library and the tags related

Is this solution seems realistic / suitable for anyone ?

**Comment by AlexisDrogoul** [comment]
I've linked (and closed) Issue #1403 with this one.

**Comment by mazarsju** [comment]
@dphilippon what is the state of this issue ?

**Comment by AlexisDrogoul** [comment]
I close this issue as Damien has finished the documentation. The "features" request is handled by issue #1581
